packages/fela-dom/src/dom/rehydration/rehydrateRules.js

```diff
@@ -9,6 +9,21 @@ function escapeRegExp(string) {
   return string.replace(/[.*+\-?^${}()|[\]\\]/g, '\\$&')
 }
 
+// we reuse regexes for improved performance
+const regexMap = {}
+function getRegex(specificityPrefix) {
+  if (!regexMap[specificityPrefix]) {
+    regexMap[specificityPrefix] = new RegExp(
+      `${escapeRegExp(
+        specificityPrefix
+      )}[.]([0-9a-z_-]+)([.][0-9a-z_-]+){0,}([^{]+)?{([^:]+):([^}]+)}`,
```

Comment:
I think you could slightly simplify it like this:
```suggestion
      )}[.]([\w-]+)([.][\w-]+){0,}([^{]+)?{([^:]+):([^}]+)}`,
```